plugins/nexusoperations/tasks.go

```diff
@@ -0,0 +1,245 @@
+// The MIT License
+//
+// Copyright (c) 2024 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package nexusoperations
+
+import (
+	"fmt"
+	"time"
+
+	"go.temporal.io/server/service/history/hsm"
+	"go.temporal.io/server/service/history/queues"
+)
+
+var (
+	TaskTypeTimeout = hsm.TaskType{
+		ID:   3,
+		Name: "nexusoperations.Timeout",
+	}
+	TaskTypeInvocation = hsm.TaskType{
+		ID:   4,
+		Name: "nexusoperations.Invocation",
+	}
+	TaskTypeBackoff = hsm.TaskType{
+		ID:   5,
+		Name: "nexusoperations.Backoff",
+	}
+	TaskTypeCancelation = hsm.TaskType{
+		ID:   6,
+		Name: "nexusoperations.Cancelation",
+	}
+	TaskTypeCancelationBackoff = hsm.TaskType{
+		ID:   7,
+		Name: "nexusoperations.CancelationBackoff",
+	}
+)
+
+type TimeoutTask struct {
+	Deadline time.Time
+}
+
+var _ hsm.Task = TimeoutTask{}
+
+func (TimeoutTask) Type() hsm.TaskType {
+	return TaskTypeTimeout
+}
```

Comment:
I mentioned this in a comment on a merged PR, but I do think the code would read even more nicely with `fsm` in place of `hsm`. With `fsm.TaskType`, people's brains will immediately register "state machine", and they will immediately move on to think about the fact that our state machines have Task Types, and that they generate tasks, etc. In contrast, with `hsm.TaskType`, since no part of it is familiar, it makes it harder for newcomers and a slight increase in mental friction for everyone. I don't think there's a downside -- I don't think anyone is going to stumble over the fact that our `fsm`s can be hierarchical. (Which highlights another advantage of the `fsm` name -- they don't have to be hierarchical)